src/lib/openseadragon-helpers.ts

```diff
@@ -0,0 +1,239 @@
+    }
+
+    if (svg) {
+      addSvgOverlay(viewer, svg, configOptions, scale);
+    }
+  });
+}
+
+export function createOpenSeadragonRect(
+  canvas: CanvasNormalized,
+  parsedAnnotationTarget: ParsedAnnotationTarget,
+  zoomLevel: number,
+) {
+  let x,
+    y,
+    w = 40,
+    h = 40;
+
+  if (parsedAnnotationTarget.rect) {
+    x = parsedAnnotationTarget.rect.x;
+    y = parsedAnnotationTarget.rect.y;
+    w = parsedAnnotationTarget.rect.w;
+    h = parsedAnnotationTarget.rect.h;
+  }
+
+  if (parsedAnnotationTarget.point) {
+    x = parsedAnnotationTarget.point.x;
+    y = parsedAnnotationTarget.point.y;
+  }
+
+  // TODO: How to handle SVG where no rect or point exists?
+  // @ts-ignore
+  if (parseAnnotationTarget.svg) {
+  }
+
+  const scale = 1 / canvas.width;
+  const rect = new OpenSeadragon.Rect(
+    x * scale - ((w * scale) / 2) * (zoomLevel - 1),
+    y * scale - ((h * scale) / 2) * (zoomLevel - 1),
+    w * scale * zoomLevel,
+    h * scale * zoomLevel,
+  );
+
+  return rect;
+}
+
+/**
+ * Add a rectangular overlay to an OpenSeadragon viewer
+ */
+function addRectangularOverlay(
+  viewer: OpenSeadragon.Viewer,
+  x: number,
+  y: number,
+  w: number,
+  h: number,
+  configOptions: ViewerConfigOptions,
+): void {
+  const rect = new OpenSeadragon.Rect(x, y, w, h);
+  const div = document.createElement("div");
+
+  if (configOptions.annotationOverlays) {
+    const { backgroundColor, opacity, borderType, borderColor, borderWidth } =
+      configOptions.annotationOverlays;
+
+    div.style.backgroundColor = backgroundColor as string;
+    div.style.opacity = opacity as string;
+    div.style.border = `${borderType} ${borderWidth} ${borderColor}`;
+    div.className = "annotation-overlay";
+  }
+
+  viewer.addOverlay(div, rect);
+}
+
+function convertSVGStringToHTML(svgString) {
+  if (!svgString) return null;
+  const template = document.createElement("template");
+  template.innerHTML = svgString.trim();
+  const result = template.content.children;
+
+  return result[0];
+}
+
+export function addSvgOverlay(
+  viewer: any,
+  svgString: string,
+  configOptions: ViewerConfigOptions,
+  scale: number,
+) {
+  const svgEl = convertSVGStringToHTML(svgString);
+  if (svgEl) {
+    for (const child of svgEl.children) {
+      svg_processChild(viewer, child, configOptions, scale);
+    }
+  }
+}
+
+function svg_processChild(
+  viewer: any,
+  child: ChildNode,
+  configOptions: ViewerConfigOptions,
+  scale: number,
+) {
+  if (child.nodeName === "#text") {
+    svg_handleTextNode(child);
+  } else {
+    const newElement = svg_handleElementNode(child, configOptions, scale);
+    const overlay = OsdSvgOverlay(viewer);
+    overlay.node().append(newElement);
+    overlay._svg?.setAttribute("class", "annotation-overlay");
+
+    // BUG: svg with children elements aren't formated correctly.
+    child.childNodes.forEach((child) => {
+      svg_processChild(viewer, child, configOptions, scale);
+    });
+  }
+}
```

Comment:
Should we have an `svg-helpers` file for all of these prefixed functions? I understand that they likely will only be used for openseadragon, but just curious here.